### ISSUE TITLE
`default_box=True` prevents accessing `box_safe_prefix`ed items

### ISSUE THREAD

**Issue by Pymancer** [cdgriffith/Box#141]
```python
>>> from box import Box
>>> b1 = Box({'a': {'from': {'b':1}}})
>>> b1.a.xfrom
<Box: {'b': 1}>
>>> b3 = Box({'a': {'from': {'b':1}}}, default_box=True)
>>> b3.a.xfrom
<Box: {}>
>>> import box
>>> box.__version__
'4.2.1'
```

**Comment by cdgriffith** [comment]
Without looking at code my guess is this is because it does a quick litterally lookup for that key and doesn't find it, so automatically gives back the new default box.  

I am not sure if this behavior should happen at all for already safe attributes like exampled though, will need to look into that as well.

**Comment by cdgriffith** [comment]
Addressing in 4.2.2 https://github.com/cdgriffith/Box/pull/142

**Comment by cdgriffith** [comment]
Fixed in 4.2.2
